### ISSUE TITLE
Enabling "with-axum" feature on askama causes build errors

### ISSUE THREAD

**Issue by qwandor** [djc/askama#588]
If I enable the `with-axum` feature in my `Cargo.toml` like

```toml
askama = {version = "0.11.0", features = ["with-axum"] }
```

then I get lots of build errors like

```
error[E0433]: failed to resolve: could not find `askama_axum` in the list of imported crates
   --> src/controllers/users.rs:134:10
    |
134 | #[derive(Template)]
    |          ^^^^^^^^ could not find `askama_axum` in the list of imported crates
    |
    = note: this error originates in the derive macro `Template` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Maybe this is because the `askama_axum` crate has not been published to crates.io?

**Comment by djc** [comment]
When using the integration you should depend on askama_axum directly. It re-exports the contents of the Askama crate.

**Comment by qwandor** [comment]
I see, it still seems to missing on crates.io though. And why does https://djc.github.io/askama/integrations.html say to use the `with-axum` feature in that case?

**Comment by djc** [comment]
Ah sorry, the release is waiting for #583 and the docs are probably outdated. 😔

**Comment by EdmundsEcho** [comment]
In reviewing this issue, it seems that the dependent issues have been resolved.  Is that right?

I ask because I'm having a tough time resolving how a working example from`axum` uses the dependency: `askama = "0.11"`.  However, when I tried to build on the work, I started to get errors requesting `askama_axum`.  The docs indicate a need to use  features `with-axum`; not what the working example uses.

Finally, one of the posts in this issues suggests the documentation is *not* current.  

Thus, the request for clarification and specifically how to use this template engine using `axum`.

Thank you!!

**Comment by djc** [comment]
You should depend on askama_axum only, this should take care of any feature flags that need to be enabled. askama_axum re-exports all the symbols from askama, so you should usually not need to depend on anything else explicitly.

**Comment by EdmundsEcho** [comment]
@djc Thank you for your prompt reply.  This helps streamline and focus my debugging efforts.

And, thank you for your work here.  It's a pleasure to work with these well-thought-out packages.

\- E

**Comment by EdmundsEcho** [comment]
Agree.

Quick follow-up.  I'm getting the following error 

```rust
#[derive(askama_axum::Template)]
    |          ^^^^^^^^^^^^^^^^^^^^^ not found in `askama`
```

... from the following code:
```rust
use askama_axum::Template;
...

#[derive(askama_axum::Template)]
#[template(path = "www/index.html")]
struct WwwTemplate {}
```

**Comment by djc** [comment]
Please open a new issue for this. Ideally have a look at the generated code, paste it into your original code (and comment out the derive) to figure out what's actually wrong.

**Comment by Kijewski** [comment]
> I have not seen syntax that starts with `::`, I did a quick check and could not find anything. Why are you using `::` as a prefix here?

It's a [qualified path](https://doc.rust-lang.org/reference/paths.html#qualified-paths). That means even your code shadows the name askama  (e.g. with `mod askama {}`), the right crate is used. It's mostly used in macros, because outside of macros you know if you shadowed the identifier or not.

**Comment by thomaseizinger** [comment]
> When using the integration you should depend on askama_axum directly. It re-exports the contents of the Askama crate.

I find this to be an odd way of integrating with it. As a user, I would have expected to always only depend on the `askama` crate and activate officially supported integrations via feature-flags. Whether or not those integrations are happening through `cfg`s and optional dependencies or are separated out into another crate seems like an implementation detail to me.

What is the reason for doing it this way around?

**Comment by djc** [comment]
That was originally how it worked, but it had a lot of issues: given the broad integration support in Askama, there were conflicts in the dependencies of different web frameworks. The current implementation is actually much cleaner even if you might find it unintuitive.

**Comment by thomaseizinger** [comment]
Interesting, thanks for sharing!

**Comment by DGotshalk** [comment]
Hello, 
I've ran into issues learning rust and using axum, and I ran into the same issues where I had the 
`askama = {version = "0.12.0", features = ["with-axum"] }`
In my Cargo.toml, and got the "error[E0433]: failed to resolve: could not find `askama_axum` in the list of imported crates"  

From what I am understanding from your response, If I am using askama on an axum app, I should just have: 
`askama_axum = "0.3.0"`
In my Cargo.toml file instead? If so, then https://djc.github.io/askama/integrations.html is still outdated. 

If the latter is the case, and I should only have the askama_axum crate, what would I do if I wanted other features from askama, like mime or markdown, etc? What would go into my Cargo.toml. 

I do notice that if I don't use the `features = ["with-axum"]`, the app seems to work, but if you specifically made integration with this other package, it would seem to me that I should use the integration.

**Comment by djc** [comment]
Yes, you should depend on `askama_axum` directly. Note that it contains a whole bunch of features itself that forward to the relevant askama features. Would you be able to submit a PR for the integration docs?

**Comment by DGotshalk** [comment]
Hello,
I just got back to this!
Just to be clear:
If I am using askama with axum, My Cargo.Toml file should be:
`[dependencies]
askama = { version = "0.12.0", features =["with-axum"] }
askama_axum = "0.3.0"`

So, along with the `features=["with-axum"]` I need to also add the crate `askama_axum`. Both are required correct? If so I can update the docks with that information.

**Comment by djc** [comment]
You should be able to depend only on askama_axum, it re-exports `Template`.

**Comment by DGotshalk** [comment]
Alright, just tried it. In my Cargo.toml: 
```
[package]
name = "distraction-tracker"
version = "0.1.0"
edition = "2021"
authors = ["Drew Gotshalk"]

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
askama_axum = "0.3.0"
axum = "0.6.11"
serde = { version = "1.0.155", features = ["derive"] }
serde_json = "1.0.94"
tokio = { version = "1.26.0", features = ["macros","rt-multi-thread"] }
tower-http = { version = "0.4.0", features = ["full"] }

```


in my .rs file in the src directory:
```use askama_axum::Template;

#[derive(Template)]
#[template(path = "index.html")]
struct IndexTemplate {
    message: String,
}
```

When I run `cargo build` in the root directory of the cargo project:
```% cargo build
   Compiling distraction-tracker v0.1.0 (/home/dgotshalk/git/distraction-tracker)
error[E0433]: failed to resolve: could not find `askama` in the list of imported crates
  --> src/routes.rs:17:10
   |
17 | #[derive(Template)]
   |          ^^^^^^^^ could not find `askama` in the list of imported crates
   |
   = note: this error originates in the derive macro `Template` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0433]: failed to resolve: could not find `askama` in the list of imported crates
  --> src/routes.rs:17:10
   |
17 | #[derive(Template)]
   |          ^^^^^^^^ could not find `askama` in the list of imported crates
   |
   = note: this error originates in the derive macro `Template` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing this struct
   |
6  | use askama_axum::MarkupDisplay;
   |
help: if you import `MarkupDisplay`, refer to it directly
   |
17 | #[derive(Template)]
   |

error[E0433]: failed to resolve: could not find `askama` in the list of imported crates
  --> src/routes.rs:17:10
   |
17 | #[derive(Template)]
   |          ^^^^^^^^ could not find `askama` in the list of imported crates
   |
   = note: this error originates in the derive macro `Template` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing one of these items
   |
6  | use askama_axum::Result;
   |
6  | use axum::response::Result;
   |
6  | use core::fmt::Result;
   |
6  | use core::result::Result;
   |
     and 9 other candidates
help: if you import `Result`, refer to it directly
   |
17 | #[derive(Template)]
   |

error[E0433]: failed to resolve: could not find `askama` in the list of imported crates
  --> src/routes.rs:17:10
   |
17 | #[derive(Template)]
   |          ^^^^^^^^ could not find `askama` in the list of imported crates
   |
   = note: this error originates in the derive macro `Template` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing this trait
   |
6  | use askama_axum::Template;
   |
help: if you import `Template`, refer to it directly
   |
17 | #[derive(Template)]
For more information about this error, try `rustc --explain E0433`.
error: could not compile `distraction-tracker` due to 4 previous errors;
```


So it is asking me to use askama. If I put askama back into my depencies with the `with-axum` feature, I get none of these errors.

**Comment by djc** [comment]
Okay, yeah, I guess you need to have both crates in your dependencies.

**Comment by DGotshalk** [comment]
If that's what is needed, that's ok with me! Just wanted to get confirmation. Thank you for your time! I did make a pull request for the docs, specifically the https://djc.github.io/askama/integrations.html page I believe. Please let me know if I did that wrong or not!
